dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/jaxrspropagation/transports/B3MultiPropagationTestServlet.java

```diff
@@ -0,0 +1,87 @@
+import java.util.List;
+
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+import org.junit.Test;
+
+import componenttest.app.FATServlet;
+import io.openliberty.microprofile.telemetry.internal_fat.apps.jaxrspropagation.common.PropagationHeaderClient;
+import io.openliberty.microprofile.telemetry.internal_fat.apps.jaxrspropagation.common.PropagationHeaderEndpoint;
+import io.openliberty.microprofile.telemetry.internal_fat.common.spanexporter.InMemorySpanExporter;
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.baggage.BaggageEntryMetadata;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import jakarta.inject.Inject;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServletRequest;
+
+@SuppressWarnings("serial")
+@WebServlet("/b3MultiPropagation")
+public class B3MultiPropagationTestServlet extends FATServlet {
+
+    private static final String BAGGAGE_VALUE = "test.baggage.value";
+    private static final String BAGGAGE_METADATA = "test.baggage.metadata";
+
+    @Inject
+    private Tracer tracer;
+
+    @Inject
+    private InMemorySpanExporter spanExporter;
+
+    @Inject
+    private HttpServletRequest request;
+
+    @Test
+    public void testB3Propagation() throws URISyntaxException {
+        Span span = tracer.spanBuilder("test").startSpan();
+        try (Scope scope = span.makeCurrent()) {
+            Baggage baggage = Baggage.builder().put(BAGGAGE_KEY, BAGGAGE_VALUE, BaggageEntryMetadata.create(BAGGAGE_METADATA)).build();
+            baggage.makeCurrent();
+            PropagationHeaderClient client = RestClientBuilder.newBuilder().baseUri(PropagationHeaderEndpoint.getBaseUri(request)).build(PropagationHeaderClient.class);
+            client.get();
+        } finally {
+            span.end();
+        }
+
+        List<SpanData> spanData = spanExporter.getFinishedSpanItems(3);
+
+        SpanData testSpan = spanData.get(0);
+        SpanData clientSpan = spanData.get(1);
+        SpanData serverSpan = spanData.get(2);
+
+        // Assert correct parent-child links
+        // Shows that propagation occurred
+        assertEquals("client parent span id", testSpan.getSpanId(), clientSpan.getParentSpanId());
+        assertEquals("server parent span id", clientSpan.getSpanId(), serverSpan.getParentSpanId());
+
+        // B3 does not support baggage, so baggage should not be propagated
+        assertNull("baggage value propagated", serverSpan.getAttributes().get(BAGGAGE_VALUE_ATTR));
+        assertNull("baggage metadata propagated", serverSpan.getAttributes().get(BAGGAGE_METADATA_ATTR));
+
+        // Assert that the expected headers were used
+        assertThat(serverSpan.getAttributes().get(PropagationHeaderEndpoint.PROPAGATION_HEADERS_ATTR),
+                   containsInAnyOrder("X-B3-TraceId", "X-B3-SpanId", "X-B3-Sampled"));
```

Comment:
Might be a good idea to make these Strings into constants in PropagationHeaderEndpoint

Reply:
They're only used in two places - the list of all headers in `PropagationHeaderEndpoint` and here. I could reuse them, but I'd need an individual constant for each header and I think it's more helpful to directly see the exact names the tests are looking for.

Also note that there's one more B3 header that is checked for by `PropagationHeaderEndpoint` but isn't used by opentelemetry - `X-B3-ParentSpanId`. I think this is because the way zipkin manages spans is different to open telemetry. OTel always has one span for the client making the request and another for the server receiving it, but that doesn't seem to be the case for zipkin so they have to also send the parent span id.